fix: ordered list alignment in safari

jnm2377 opened the pull request:
Closes #985 
 
- adds webkit-padding for safari only
- **testing**: view in Safari browser, compare localhost/preview to existing list example on live site and see alignment has changed: https://gatsby-theme-carbon.now.sh/components/markdown#lists

vpicone commented on packages/gatsby-theme-carbon/src/components/markdown/Markdown.module.scss:
```diff
@@ -85,6 +85,7 @@ ul :global(.bx--list--unordered) {
 
 :global(.bx--list--ordered:not(.bx--list--nested)) {
   margin-left: 1ch;
+  -webkit-padding-start: 0.5ch; // needed to align ordered lists to body copy in Safari
```
```suggestion
  -webkit-padding-start: 0.5ch; // needed to align ordered lists to body copy in Safari #985
```

vpicone reviewed (approved):
Awesome, adding the issue numbers can be helpful I think, especially now that VS Code let's you open them from the editor

vpicone reviewed (changes requested):
![IMG_F58B955E6D95-1](https://user-images.githubusercontent.com/4078018/92804838-3cd14d00-f37e-11ea-973c-eeec2e6c5291.jpeg)

It looks like the nested list items could use this as well, maybe update the rule at line 75 to be:

```
:global(.bx--list--ordered) {
  list-style: decimal;
  -webkit-padding-start: 0.5ch; // needed to align ordered lists to body copy in Safari #985
}
```

Not sure why the `, ol` was there but looks like it can be removed.

jnm2377 commented:
@vpicone looks like the nested list alignment was off on chrome too, so I updated it for both. I also changed the way I was targeting the safari only fixes bc for some reason it was affecting chrome too. Should be good now.

vpicone commented on packages/gatsby-theme-carbon/src/components/markdown/Markdown.module.scss:
```diff
@@ -106,6 +106,21 @@ ul :global(.bx--list--unordered) {
   margin-left: 1ch;
 }
 
+:global(.bx--list--ordered.bx--list--nested) {
```
@jnm2377 Do you mind maybe moving the comment up hear and making it a little more descriptive? I think this markdown style sheet is getting buck wild and I won't remember what happened in a month haha.
